source/analysis/AM1DNormalizationAB.h

```diff
@@ -130,6 +130,8 @@ protected slots:
 
 	/// Flag for knowing whether we need to compute the values.
 	mutable bool cacheUpdateRequired_;
+	/// A list of indices of values that need to be updated.  These are meant for when scans are running and data is coming in a pixel at a time.
+	mutable QList<AMnDIndex> dirtyIndices_;
```

Comment:
Are there any analysis blocks which don't use this caching method, and as such don't require the dirty indices and cached data?

It seems that these reappear in a lot of analysis blocks, so it makes sense for them to be in a common base (either AMAnalysisBlock, AMStandardAnalysisBlock or a new AMCachedAnalysisBlock).